integration/src/main/java/scripts/testscripts/ControlledApplicationPrivateGcsBucketLifecycle.java

```diff
@@ -181,33 +199,25 @@ private void testAssignedWriter(ControlledGcpResourceApi resourceApi, String pro
             bucketResourceName,
             CloningInstructionsEnum.NOTHING,
             privateUser);
-    bucketName = createdBucket.getGcpBucket().getAttributes().getBucketName();
+    bucketList.add(createdBucket);
+    String bucketName = createdBucket.getGcpBucket().getAttributes().getBucketName();
     assertNotNull(bucketName);
     logger.info("Created assigned-writer bucket {}", bucketName);
 
+    // Creating the tester wait for wsmapp to have EDITOR permissions
     try (GcsBucketAccessTester tester = new GcsBucketAccessTester(wsmapp, bucketName, projectId)) {
-      tester.checkAccess(wsmapp, ControlledResourceIamRole.EDITOR);
-      tester.checkAccess(writer, null);
-      tester.checkAccess(reader, ControlledResourceIamRole.WRITER);
+      tester.assertAccess(writer, null);
+      tester.assertAccessWait(reader, ControlledResourceIamRole.WRITER);
     }
-    deleteBucket(resourceApi, createdBucket);
-  }
-
-  private void deleteBucket(
-      ControlledGcpResourceApi resourceApi, CreatedControlledGcpGcsBucket createdBucket)
-      throws Exception {
-    GcsBucketUtils.deleteControlledGcsBucket(
-        createdBucket.getResourceId(), getWorkspaceId(), resourceApi);
-    logger.info("Application deleted bucket {}", bucketName);
-    bucketName = null;
   }
 
-  @Override
-  protected void doCleanup(List<TestUserSpecification> testUsers, WorkspaceApi workspaceApi)
-      throws Exception {
-    super.doCleanup(testUsers, workspaceApi);
-    if (bucketName != null) {
-      logger.warn("Test failed to cleanup bucket " + bucketName);
+  private void deleteBucketList() throws Exception {
+    for (CreatedControlledGcpGcsBucket bucket : bucketList) {
+      GcsBucketUtils.deleteControlledGcsBucket(
+          bucket.getResourceId(), getWorkspaceId(), wsmappResourceApi);
+      logger.info(
+          "Application deleted bucket {}", bucket.getGcpBucket().getAttributes().getBucketName());
     }
+    bucketList.clear();
```

Comment:
if you clear this in doUserJourney, there's nothing to delete in cleanUp()

Reply:
The idea is that on a successful test run, the deletes happen as part of the test. So we notice a failure if they fail.
On some other test failure, we will still delete any buckets we have made so far. Do you think that works?